tests/test_core.py

```diff
@@ -1859,76 +1869,64 @@ class TestO:
 
         self.assertIsNone(wr_python_object())
 
-    def test_objcinstance_release_owned(self):
-        # Create an object which we own.
-        obj = NSObject.alloc().init()
-
-        # Check that it is marked for release.
-        self.assertTrue(obj._needs_release)
-
-        # Explicitly release the object.
-        obj.release()
-
-        # Check that we no longer need to release it.
-        self.assertFalse(obj._needs_release)
+    def test_objcinstance_created_retained(self):
+        with autoreleasepool():
+            # Create an object which we don't own.
+            NSString = ObjCClass("NSString")
+            obj = NSString.stringWithString(str(uuid.uuid4()))
 
-        # Delete it and make sure that we don't segfault on garbage collection.
-        del obj
-        gc.collect()
+        self.assertEqual(obj.retainCount(), 1)
 
-    def test_objcinstance_autorelease_owned(self):
+    def test_objcinstance_explicit_retained(self):
         # Create an object which we own.
         obj = NSObject.alloc().init()
 
-        # Check that it is marked for release.
-        self.assertTrue(obj._needs_release)
-
-        # Explicitly release the object.
-        res = obj.autorelease()
+        self.assertEqual(obj.retainCount(), 1)
 
-        # Check that autorelease call returned the object itself.
-        self.assertIs(obj, res)
+    def test_objcinstance_created_gc_released(self):
+        # Create an object which we own.
+        obj = NSObject.alloc().init()
 
-        # Check that we no longer need to release it.
-        self.assertFalse(obj._needs_release)
+        wr = ObjcWeakref.alloc().init()
+        wr.weak_property = obj
 
         # Delete it and make sure that we don't segfault on garbage collection.
-        del obj
-        gc.collect()
-
-    def test_objcinstance_retain_release(self):
-        NSString = ObjCClass("NSString")
+        with autoreleasepool():
+            del obj
+            gc.collect()
 
-        # Create an object which we don't own.
-        string = NSString.stringWithString("test")
+        # Assert that the obj was deallocated.
+        self.assertIsNone(wr.weak_property)
```

Comment:
We should also verify that the object is *not* deallocated between `gc.collect` and the end of the `autoreleasepool`. This distinguishes between `release` and `autorelease`.

Reply:
Good point. Done.